### ISSUE TITLE
Lack of ws.Message (un)marshaller

### ISSUE THREAD

**Issue by ipsq** [akka/akka-http#2256]
Shouldn't there exist a standard ws.Message FromWSMessageUnmarshaller to basic types such as String and ByteString. Even for JSON and XML, support for FromWSMessageUnmarshaller[T] or even FromWSMessageUnmarshaller[JsValue] would come in handy since these are scattered across multiple codebases whilst I believe they should be part of the project itself. I'm willing to submit a PR if this is approved.

**Comment by raboof** [comment]
I agree it would be good to have an easy way to unmarshall websocket messages.

Ideally it would compose nicely with the existing marshalling/unmarshalling support. For example, for JSON, would it make sense to introduce a `FromWsMessageUnmarshaller` next to `FromEntityUnmarshaller` and add support for it to e.g. `SprayJsonSupport`? Or would another approach be better?
